src/test/java/io/cdap/plugin/ariba/source/AribaBatchSourceTest.java

```diff
@@ -84,23 +84,39 @@ public void testConfigurePipelineWithInvalidBasicParam() {
       aribaBatchSource.configurePipeline(pipelineConfigurer);
     } catch (ValidationException ve) {
       List<ValidationFailure> failures = ve.getFailures();
-      //Reference name throws 2 failure, one from our method and another from CDAP Util class.
-      //Other 4 properties are kept blank
-      Assert.assertEquals("Failures size does not match", 8, failures.size());
-      Assert.assertEquals(ResourceConstants.ERR_MISSING_PARAM_PREFIX.getMsgForKey("Reference Name"),
-                          failures.get(0).getMessage());
+      Assert.assertEquals("Failures size does not match", 6, failures.size());
 
       Assert.assertEquals(ResourceConstants.ERR_MISSING_PARAM_PREFIX.getMsgForKey("API Endpoint"),
-                          failures.get(1).getMessage());
+                          failures.get(0).getMessage());
 
       Assert.assertEquals(ResourceConstants.ERR_MISSING_PARAM_PREFIX.getMsgForKey("System Type"),
-                          failures.get(2).getMessage());
+                          failures.get(1).getMessage());
 
       Assert.assertEquals(ResourceConstants.ERR_MISSING_PARAM_PREFIX.getMsgForKey("Realm Name"),
-                          failures.get(3).getMessage());
+                          failures.get(2).getMessage());
 
       Assert.assertEquals(ResourceConstants.ERR_MISSING_PARAM_PREFIX.getMsgForKey("View Template Name"),
-                          failures.get(4).getMessage());
+                          failures.get(3).getMessage());
+
+    }
+  }
+
+  @Test
+  public void testConfigurePipelineWithEmptyReferenceName() {
+    pluginConfig = new AribaPluginConfig("", "url",
+                                         "sysType", "realm",
+                                         "template", "clientId",
+                                         "clientSecret", "apiKey",
+                                         "2022-01-28T10:05:02Z", "2022-01-31T10:05:02Z");
+    try {
+      aribaBatchSource = new AribaBatchSource(pluginConfig);
+      aribaBatchSource.configurePipeline(pipelineConfigurer);
+    } catch (ValidationException ve) {
+      List<ValidationFailure> failures = ve.getFailures();
+      //Reference name throws 2 failure, one from our method and another from CDAP Util class.
```

Comment:
what is the second error related to "CDAP Util class"?

Reply:
Invalid reference name.